packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -1376,3 +1380,165 @@ export async function t9n(componentTestSetup: ComponentTestSetup): Promise<void>
     await page.waitForChanges();
   }
 }
+
+/**
+ * Helper to test openClose component setup.
+ *
+ * Note that this helper should be used within a `describe` block.
+ *
+ * @example
+ * import { openClose } from "../../tests/commonTests";
+ *
+ * openClose("calcite-tooltip", "open", false, {
+ *   open: async (page) => {
+ *     await page.keyboard.press("Tab");
+ *     await page.waitForChanges();
+ *   },
+ *   close: async (page) => {
+ *     await page.keyboard.press("Tab");
+ *     await page.waitForChanges();
+ *   },
+ * });
+ *
+ * @param {string} componentTagOrHTML - the component tag or HTML markup to test against
+ * @param {string} toggleProp - Toggle property to test. Currently, either "open" or "expanded".
+ * @param {boolean} initialToggleValue - Indicates the initial value of the toggle property to determine whether to configure a `simplePageSetup` or `newProgrammaticE2EPage`.
+ * @param {userInputDevice} userInputDevice - Optional argument with functions to simulate user input (mouse or keyboard), to open or close the component.
+ */
+
+type userInputDevice = {
```

Comment:
Please add JSDoc to the interface similar to other test util options. Applies to all test-helper-related types/interfaces.